config/main.py

```diff
@@ -1893,36 +1893,37 @@ def override_config_table(db, input_config_db, dry_run):
                     fg='magenta')
         sys.exit(1)
 
-    config_db = db.cfgdb
-
-    # Read config from configDB
-    current_config = config_db.get_config()
-    # Serialize to the same format as json input
-    sonic_cfggen.FormatConverter.to_serialized(current_config)
-
-    updated_config = update_config(current_config, config_input)
-
-    yang_enabled = device_info.is_yang_config_validation_enabled(config_db)
-    if yang_enabled:
-        # The ConfigMgmt will load YANG and running
-        # config during initialization.
-        try:
-            cm = ConfigMgmt()
-            cm.validateConfigData()
-        except Exception as ex:
-            click.secho("Failed to validate running config. Error: {}".format(ex), fg="magenta")
-            sys.exit(1)
+    # Do the yang validation and config override for host namespace and 
+    # other namespaces in case of multi-asic platform
+    for ns, config_db in db.cfgdb_clients.items():
```

Comment:
>for [](http://example.com/codeflow?start=4&length=3)

I understand that `MACSEC_PROFILE` will be in all namespaces in addition to the host namespace. However this is right in general.

Can we limit the loop only for `MACSEC_PROFILE` table?

Reply:
I have updated the logic to have a list of tables which is there per namespace, if any of the tables in the config_input json is present in that list, we do override config for all namespaces. Else only for host. Currently MACSEC_PROFILE is there in that table list.

this logic can be generalized to any table depending on how we plan to go with Virtual connector, to do this action based on table, and have it give back which connector to use, or which namespace the config needs to be overridden.